src/targeting.js

```diff
@@ -0,0 +1,181 @@
+  }
+};
+
+targeting.getAllTargeting = function(adUnitCode) {
+  const adUnitCodes = adUnitCode && adUnitCode.length ? [adUnitCode] : $$PREBID_GLOBAL$$._adUnitCodes;
+
+  // Get targeting for the winning bid. Add targeting for any bids that have
+  // `alwaysUseBid=true`. If sending all bids is enabled, add targeting for losing bids.
+  var targeting = getWinningBidTargeting(adUnitCodes)
+      .concat(getAlwaysUseBidTargeting(adUnitCodes))
+      .concat($$PREBID_GLOBAL$$._sendAllBids ? getBidLandscapeTargeting(adUnitCodes) : [])
+      .concat(getDealTargeting(adUnitCodes));
+
+  //store a reference of the targeting keys
+  targeting.map(adUnitCode => {
+    Object.keys(adUnitCode).map(key => {
+      adUnitCode[key].map(targetKey => {
+        if (pbTargetingKeys.indexOf(Object.keys(targetKey)[0]) === -1) {
+          pbTargetingKeys = Object.keys(targetKey).concat(pbTargetingKeys);
+        }
+      });
+    });
+  });
+  return targeting;
+};
+
+targeting.setTargeting = function(targetingConfig) {
+  window.googletag.pubads().getSlots().forEach(slot => {
+    targetingConfig.filter(targeting => Object.keys(targeting)[0] === slot.getAdUnitPath() ||
+      Object.keys(targeting)[0] === slot.getSlotElementId())
+      .forEach(targeting => targeting[Object.keys(targeting)[0]]
+        .forEach(key => {
+          key[Object.keys(key)[0]]
+            .map((value) => {
+              utils.logMessage(`Attempting to set key value for slot: ${slot.getSlotElementId()} key: ${Object.keys(key)[0]} value: ${value}`);
+              return value;
+            })
+            .forEach(value => {
+              slot.setTargeting(Object.keys(key)[0], value);
+            });
+        }));
+  });
+};
+
+targeting.getWinningBids = function(adUnitCode) {
+  // use the given adUnitCode as a filter if present or all adUnitCodes if not
+  const adUnitCodes = adUnitCode ? [adUnitCode] : $$PREBID_GLOBAL$$._adUnitCodes;
+
+  return $$PREBID_GLOBAL$$._bidsReceived
+    .filter(bid => adUnitCodes.includes(bid.adUnitCode))
+    .filter(bid => bid.cpm > 0)
+    .map(bid => bid.adUnitCode)
+    .filter(uniques)
+    .map(adUnitCode => $$PREBID_GLOBAL$$._bidsReceived
+      .filter(bid => bid.adUnitCode === adUnitCode ? bid : null)
+      .reduce(getHighestCpm,
+        {
+          adUnitCode: adUnitCode,
+          cpm: 0,
+          adserverTargeting: {},
+          timeToRespond: 0
+        }));
+};
+
+targeting.setTargetingForAn = function() {
+  let targeting = $$PREBID_GLOBAL$$.getAdserverTargeting();
+  Object.keys(targeting).forEach(targetId =>
+    Object.keys(targeting[targetId]).forEach(key => {
+      utils.logMessage(`Attempting to set targeting for targetId: ${targetId} key: ${key} value: ${targeting[targetId][key]}`);
+      //setKeywords supports string and array as value
+      if(utils.isStr(targeting[targetId][key]) || utils.isArray(targeting[targetId][key])) {
+        var keywordsObj = {};
+        keywordsObj[key] = targeting[targetId][key];
```

Comment:
We also need to multiple the value of `hb_pb` by 100 and trim it (no decimals). `(value *  100).toFixed();`